src/mlpack/methods/reinforcement_learning/replay/episodic_replay.hpp

```diff
@@ -0,0 +1,271 @@
+
+
+  /**
+  * Store the given experience.
+  * @param state Given state.
+  * @param action Given action.
+  * @param reward Given reward.
+  * @param nextState Given next state.
+  * @param isEnd Whether next state is terminal state.
+  */
+  void Store(const StateType& state,
+             ActionType action,
+             double reward,
+             const StateType& nextState,
+             bool isEnd)
+  {
+    if(clear)
+    {
+      states[position].clear();
+      actions[position].clear();
+      rewards[position].clear();
+      next_states[position].clear();
+      isTerminal[position].clear();
+      clear = false;
+    }
+    states[position].push_back(state.Encode());
+    actions[position].push_back(action);
+    rewards[position].push_back(reward);
+    next_states[position].push_back(nextState.Encode());
+    isTerminal[position].push_back(isEnd);
+    if(isEnd||states[position].size()==max_episode_len)
+    {
+      position++;
+      clear = true;
+    }
+    if (position == capacity)
+    {
+      full = true;
+      position = 0;
+    }
+  }
+
+  /**
+  * Sample some episode.
+  * @param sampledStates Sampled encoded states.
+  * @param sampledActions Sampled actions.
+  * @param sampledRewards Sampled rewards.
+  * @param sampledNextStates Sampled encoded next states.
+  * @param isTerminal Indicate whether corresponding next state is terminal
+  *        state.
+  */
+
+  void Random_Episode(arma::mat& episodeStates,
+                      arma::icolvec& episodeActions,
+                      arma::colvec& episodeRewards,
+                      arma::mat& episodeNextStates,
+                      arma::icolvec& isTerminal)
+  {
+    size_t upperBound = full ? capacity : position;
+    int lo = 0;
+    int high = upperBound;
+    if(!upperBound)
+    {
+      high = 1;
+    }
+    int episodeNum = math::RandInt(lo,high);
+    std::vector<arma::colvec> temp = states[episodeNum];
+    int i = 0;
+    for(auto state : temp)
+    {
+      if(i==0)
+      {
+        episodeStates = state;
+        i++;
+      }else
+      {
+        episodeStates = arma::join_rows(episodeStates, state);
+      }
+    }
+    episodeActions = arma::conv_to<arma::icolvec>::from(actions[episodeNum]);
+    episodeRewards = arma::conv_to<arma::colvec>::from(rewards[episodeNum]);
+    temp = next_states[episodeNum];
+    i = 0;
+    for(auto state : temp)
+    {
+      if(i==0)
+      {
+        episodeNextStates = state;
+        i++;
+      }else
+      {
+        episodeNextStates = arma::join_rows(episodeNextStates, state);
+      }
+    }
+    isTerminal = arma::conv_to<arma::icolvec>::from(this->isTerminal[episodeNum]);
+    }
+
+  /**
+  * Get the number of episodes in the memory.
+  *
+  * @return Actual used memory size
+  */
+
+  const size_t Size()
+  {
+    if(states[position].size()==0){
+      return full ? capacity : position;  
+    }else{
+      return full ? capacity : (position+1);  
+    }
+    return full ? capacity : position;
```

Comment:
Looks like we can use `return full ? capacity : (position+1); ` here and remove the else case.